docker-compose.yml

```diff
@@ -0,0 +1,111 @@
+version: '3.5'
+services:
+  app: &app
+    build:
+      context: .
+      target: hyrax-engine-dev
+      args:
+        BUNDLE_WITHOUT: test
+    image: hyrax-engine-dev
+    stdin_open: true
+    tty: true
+    environment:
+      - DATABASE_URL=postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5
+      - FCREPO_PORT=8080
+      - FCREPO_HOST=fcrepo
+      - FCREPO_REST_PATH=fcrepo/rest
+      - HYRAX_ENGINE_PATH=/app-data/samvera/hyrax-engine
+      - RACK_ENV=development
+      - RAILS_ENV=development
+      - REDIS_HOST=redis
+      - SOLR_URL=http://solr:8983/solr/hyrax
+    depends_on:
+      - db_migrate
+      - fcrepo
+      - memcached
+      - postgres
+      - redis
+      - solr
+    ports:
+      - 3000:3000
+    volumes:
+      - .dassie:/app-data/samvera/hyrax-webapp
+      - .:/app-data/samvera/hyrax-engine
+
+  db_migrate:
+    image: hyrax-engine-dev
+    environment:
+      - DATABASE_URL=postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5
+      - FCREPO_PORT=8080
+      - FCREPO_HOST=fcrepo
+      - FCREPO_REST_PATH=fcrepo/rest
+      - HYRAX_ENGINE_PATH=/app-data/samvera/hyrax-engine
+      - SOLR_URL=http://solr:8983/solr/hyrax
+    command:
+      - /app-data/samvera/db-wait.sh
+      - postgres:5432
+      - /app-data/samvera/db-wait.sh
+      - fcrepo:8080
+      - /app-data/samvera/db-wait.sh
+      - solr:8983
+      - bundle
+      - exec
+      - rails
+      - db:setup
+      - db:migrate
+      - db:seed
+    depends_on:
+      - postgres
+    volumes:
+      - .dassie:/app-data/samvera/hyrax-webapp
+      - .:/app-data/samvera/hyrax-engine
+
+  postgres:
+    image: postgres:latest
+    restart: always
+    environment:
+      - POSTGRES_USER=hyrax_user
+      - POSTGRES_PASSWORD=hyrax_password
+      - POSTGRES_DB=hyrax
+      - POSTGRES_HOST_AUTH_METHOD=trust
+    ports:
+      - "5432:5432"
+    volumes:
+      - db:/var/lib/postgresql/data
+
+  fcrepo:
+    image: cbeer/fcrepo4:4.7
+    volumes:
+      - fcrepo:/data
+    ports:
+      - 8080:8080
+    environment:
+      - JAVA_OPTS=${JAVA_OPTS} -Dfcrepo.modeshape.configuration="classpath:/config/file-simple/repository.json" -Dfcrepo.object.directory="/data/objects" -Dfcrepo.binary.directory="/data/binaries"
+
+  memcached:
```

Comment:
two factors here:

  - i started this work while i was doing development on parts of the code base that rely on caching. i needed a good cache configuration to test against, and memcached was that for me. causally, that's why it's here now
  - i'm hoping this configuration will result in a pretty good baseline for folks to base their production infrastructure on. i would normally roll `memcached` for the rails cache. i haven't leaned on `redis` for this purpose, historically, so i'm less likely to be prepared to support that in the community. do you think we should consider recommending folks point their rails cache at `redis`?

Reply:
I think go forward with memcached then, as it is the default.  I think that one thing that is true (and very good) about this work is it will make having a conversation about all the services needed to run Hyrax easier to have and a little more prominent.  Punting "what if we used fewer dependent services" down the road and just capturing what is common practice right now feels more than sufficient scope wise. I'll save my soap box for another time ;-)

Reply:
:+1: i think this is a great catalyst for soap-boxes.